src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -120,588 +107,213 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
-  }
+  std::variant<Variable, Alias> visit(Parser::VarOrAliasContext* ctx);
 
-  SolutionModifiers visitTypesafe(Parser::SolutionModifierContext* ctx);
+  Alias visit(Parser::AliasContext* ctx);
 
-  Any visitGroupClause(Parser::GroupClauseContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  Alias visit(Parser::AliasWithoutBracketsContext* ctx);
 
-  vector<GroupKey> visitTypesafe(Parser::GroupClauseContext* ctx);
+  ParsedQuery visit(Parser::ConstructQueryContext* ctx);
 
-  Any visitGroupCondition(Parser::GroupConditionContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  [[noreturn]] ParsedQuery visit(Parser::DescribeQueryContext* ctx);
 
-  GroupKey visitTypesafe(Parser::GroupConditionContext* ctx);
+  [[noreturn]] ParsedQuery visit(Parser::AskQueryContext* ctx);
 
-  Any visitHavingClause(Parser::HavingClauseContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  [[noreturn]] void visit(Parser::DatasetClauseContext* ctx);
 
-  vector<SparqlFilter> visitTypesafe(Parser::HavingClauseContext* ctx);
+  [[noreturn]] void visit(Parser::DefaultGraphClauseContext* ctx);
 
-  Any visitHavingCondition(Parser::HavingConditionContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  [[noreturn]] void visit(Parser::NamedGraphClauseContext* ctx);
 
-  SparqlFilter visitTypesafe(Parser::HavingConditionContext* ctx);
+  [[noreturn]] void visit(Parser::SourceSelectorContext* ctx);
 
-  Any visitOrderClause(Parser::OrderClauseContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  PatternAndVisibleVariables visit(Parser::WhereClauseContext* ctx);
 
-  vector<OrderKey> visitTypesafe(Parser::OrderClauseContext* ctx);
+  SolutionModifiers visit(Parser::SolutionModifierContext* ctx);
 
-  Any visitOrderCondition(Parser::OrderConditionContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  vector<GroupKey> visit(Parser::GroupClauseContext* ctx);
 
-  OrderKey visitTypesafe(Parser::OrderConditionContext* ctx);
+  GroupKey visit(Parser::GroupConditionContext* ctx);
 
-  Any visitLimitOffsetClauses(Parser::LimitOffsetClausesContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  vector<SparqlFilter> visit(Parser::HavingClauseContext* ctx);
 
-  LimitOffsetClause visitTypesafe(Parser::LimitOffsetClausesContext* ctx);
+  SparqlFilter visit(Parser::HavingConditionContext* ctx);
 
-  Any visitLimitClause(Parser::LimitClauseContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  vector<OrderKey> visit(Parser::OrderClauseContext* ctx);
 
-  unsigned long long int visitTypesafe(Parser::LimitClauseContext* ctx);
+  OrderKey visit(Parser::OrderConditionContext* ctx);
 
-  Any visitOffsetClause(Parser::OffsetClauseContext* ctx) override {
-    return visitTypesafe(ctx);
-  }
+  LimitOffsetClause visit(Parser::LimitOffsetClausesContext* ctx);
 
-  unsigned long long int visitTypesafe(Parser::OffsetClauseContext* ctx);
+  unsigned long long int visit(Parser::LimitClauseContext* ctx);
```

Comment:
Should we add add `static_assert` here to check that `unsigned long long int` is indeed 64bit? Theoretically it might be longer. If that were the case it might lead to overflows because we'd then be casting to a smaller type. Or even better check it in the `CMakeLists.txt`. I've played with both options locally. Let me know what you think.

Reply:
No I think we should just use a type with a well-defined length. I wrote the `unsigned long long` before I knew better, and I think we should consistently use `uint64_t`, `uint32_t`, etc. QLever internally supports at most 64 bits, sometimes even less currently (because we use some bits for tagging), so there currently is no use case. If we ever get to 128 bit machines, then we still have to rewrite a lot of things, but this will also get easier if we have fixed-size types. 

TLDR: Just change the type to `uint64_t`.

Reply:
Yes I have already applied the suggestion it. But `std:stoull` still returns `unsigned long long` and is converted to `uint64_t` regardless.